src/main/twirl/views/renderers/dateField.scala.html

```diff
@@ -1,28 +1,31 @@
-@(field: forms.DateField)
+@(field: forms.DateField, questions: Map[String, Question], answers: Map[String, String], errs: Seq[forms.validation.FieldError])
 
+@errors = @{
+    errs.filter(p => p.path == field.name || p.path.startsWith(s"${field.name}."))
+}
 
-<div class='form-group @{if (field.errs.size > 0) "error"}'>
-    @field.question.map { q => <p class="question">@q.text</p> }
-    @field.errs.map { es =>
-        @es.toList.map { e =>
-            <span class="error-message">@e</span>
-        }
+@errorClass = @{if(errors.nonEmpty) "error" else ""}
+
+<div class='form-group @errorClass'>
+    @questions.get(field.name).map { q => <p class="question">@q.text</p> }
+    @errors.map { e =>
+        <span class="error-message">@e.err</span>
     }
 
     <div class='date-selector'>
         <div>
-            <label for="@(field.name)__day">Day</label>
-            <input type="text" title="@(field.name)__day" id="@(field.name)__day" name="@(field.name)__day" value='@field.value.map(_.day)' class="form-control">
+            <label for="@(field.name).day">Day</label>
+            <input type="text" title="@(field.name).day" id="@(field.name).day" name="@(field.name).day" value='@answers.get(s"${field.name}.day")' class="form-control">
```

Comment:
I'd like to see a `pattern="[0-9]*"` attribute here to bring up the numeric keypad on mobiles.  But would then also need a `novalidate="novalidate"` attribute on the form itself to prevent the browsers from interpreting it - I  don't think we want browser based HTML5 validation at all.

I'd be happy to raise a small PR for this later on if that works better for you.